src/float.jl

```diff
@@ -0,0 +1,147 @@
+module FloatEncoding
+using Supposition: uint, tear, bias, fracsize, exposize, max_exponent, assemble
+
+"""
+    exponent_key(T, e)
+
+A lexographical ordering for floating point exponents. The encoding is taken
+from hypothesis.
+The ordering is
+- non-negative exponents in increasing order
+- negative exponents in decreasing order
+- the maximum exponent
+"""
+function exponent_key(::Type{T}, e::iT) where {T<:Base.IEEEFloat,iT<:Unsigned}
+    if e == max_exponent(T)
+        return Inf
+    end
+    unbiased = float(e) - bias(T)
+    if unbiased < 0
+        10000 - unbiased
+    else
+        unbiased
+    end
+end
+
+_make_encoding_table(T) = sort(
+    zero(uint(T)):max_exponent(T),
+    by=Base.Fix1(exponent_key, T))
+const ENCODING_TABLE = Dict(
+    UInt16 => _make_encoding_table(Float16),
+    UInt32 => _make_encoding_table(Float32),
+    UInt64 => _make_encoding_table(Float64))
+
+encode_exponent(e::T) where {T<:Unsigned} = ENCODING_TABLE[T][e+1]
+
+function _make_decoding_table(T)
+    decoding_table = zeros(uint(T), max_exponent(T) + 1)
+    for (i, e) in enumerate(ENCODING_TABLE[uint(T)])
+        decoding_table[e+1] = i - 1
+    end
+    decoding_table
+end
+const DECODING_TABLE = Dict(
+    UInt16 => _make_decoding_table(Float16),
+    UInt32 => _make_decoding_table(Float32),
+    UInt64 => _make_decoding_table(Float64))
+decode_exponent(e::T) where {T<:Unsigned} = DECODING_TABLE[T][e+1]
```

Comment:
Docstrings for these & `ENCODING_TABLE`/`encode_exponent` above, please :) A small blurb describing what the constant is used for is enough, so that a quick `help> FloatEncoding.DECODING_TABLE` is helpful.